components/brave_wallet/renderer/resource_helper.cc

```diff
@@ -18,4 +22,25 @@ std::string LoadDataResource(const int id) {
   return std::string(resource_bundle.GetRawDataResource(id));
 }
 
+absl::optional<std::string> LoadImageResourceAsDataUrl(const int id) {
+  auto& resource_bundle = ui::ResourceBundle::GetSharedInstance();
+  if (resource_bundle.IsGzipped(id)) {
+    return resource_bundle.LoadDataResourceString(id);
+  }
+
+  auto image = resource_bundle.GetImageNamed(id);
+  if (image.IsEmpty()) {
+    return absl::nullopt;
+  }
+
+  auto encoded = base::MakeRefCounted<base::RefCountedBytes>();
```

Comment:
It returns scoped_refptr. RefCountedBytes has byte array object, so it is better to keep it on the heap. We can't create unique since destructor is private, but RefCountedBytes extends RefCountedThreadSafe<RefCountedMemory>

Reply:
sorry I should have be more specific. Why do we need `RefCountedBytes` here in order to use `EncodeBGRASkBitmap`?

Reply:
This is how it is used in other places, i think just to clear data after function exits

Reply:
Not all of them and I believe the place you reference from is passing the bytes between thread. ex. `//chrome/browser/ui/webui/theme_source.cc`
Most of them don't need that, see `//chrome/browser/web_applications/os_integration/icns_encoder.cc`